connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,246 @@
+                items:
+                  type: object
+                  required:
+                    - pageId
+                    - results
+                    - logs
+                    - request
+                    - response
+                  properties:
+                    pageId:
+                      type: string
+                      description: String identifying the current page
+                    results:
+                      type: array
+                      description: The RECORD and STATE AirbyteMessages coming from the read operation for this page
+                      items:
+                        type: string
+                    logs:
+                      type: array
+                      description: The LOG AirbyteMessages coming from the read operation for this page
+                      items:
+                        type: string
+                    request:
+                      type: object
+                      description: The HTTP request sent to the source API for this page
+                      required:
+                        - url
+                        - headers
+                      properties:
+                        url:
+                          type: string
+                          description: URL that the request was sent to
+                        body:
+                          type: object
+                          description: The body of the HTTP request, if present
+                        headers:
+                          type: object
+                          description: The headers of the HTTP request
+                    response:
+                      type: object
+                      description: The response from the source API for this page
+                      required:
+                        - status
+                        - headers
+                      properties:
+                        status:
+                          type: integer
+                          description: The status of the response
+                        body:
+                          type: object
+                          description: The body of the HTTP response, if present
+                        headers:
+                          type: object
+                          description: The headers of the HTTP response
+    StreamReadRequestBody:
+      type: object
+      required:
+        - definition
+        - stream
+      properties:
+        definition:
+          $ref: "#/components/schemas/ConnectorDefinitionBody"
+          description: The config-based connector definition contents
+        stream:
+          type: string
+          description: Name of the stream to read
+        numSlices:
+          type: integer
+          description: Number of stream slices to read from the source
+          default: 1
+        numPages:
+          type: integer
+          description: Number of pages to read from the source
```

Comment:
`Number of pages to read per-slice from the source`

We should probably be very explicit in the description about what this means. Presumably we mean number of pages per slice we will retrieve. i.e. slices = 3, pages = 5 means that for each slice we'll get 5 pages worth of records.

As it reads now, it might be conflated to mean 3 total pages, but pages are kind of a subprocess of slicing